packages/core/src/dialects/mariadb/query-interface.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+import { getObjectFromMap } from '../../utils/object';
+import { assertNoReservedBind, combineBinds } from '../../utils/sql';
+
+const { AbstractQueryInterface } = require('../abstract/query-interface');
+const { QueryTypes } = require('../../query-types');
+
+/**
+ * The interface that Sequelize uses to talk with MariaDB database
+ */
+export class MariaDbQueryInterface extends AbstractQueryInterface {
```

Comment:
Is this change related to the migration of constraints to typescript or can it be moved to its own PR? That sounds like something that deserves its own focus

Reply:
The reason for this is that mariadb supports removing constraints however mysql doesn't however the mariadb dialect was set to use the MySqlQueryInterface.

I notice that MySqlQueryInterface and MariaDbQueryInterface share the same `removeColumn` and 'upsert` methods, should I update the MySqlQueryInterface to extend the MariaDbQueryInterface or leave them separate?

Reply:
Just to note on my review progress: This & the tests are the last thing I still need to review for this PR

Reply:
Ok no worries, I'll mark this conversation as resolved.